test/classes/initializers/cond-only-else-1.good

```diff
@@ -1,2 +1,2 @@
 cond-only-else-1.chpl:16: In initializer:
-cond-only-else-1.chpl:19: error: Both arms of a conditional must use this.init() or super.init() in phase 1
+cond-only-else-1.chpl:19: error: Both arms of a conditional must use this.init() or this.initDone() in phase 1
```

Comment:
This seems odd.

Reply:
The rationale behind this error is that the phase of the initializer should be the same at the end of all branches of a conditional. this.init() and initDone() both transition to phase two.

Does that make this seem less odd, or was there something else?

Reply:
I meant, this.init() and super.init() seem like peers in many senses, but the error mentions only this.init() as an option.

Reply:
In the new proposal when super.init() is called the initializer transitions to phase one, which doesn't match the phase after this.init() (phase two).

Reply:
Ah, ok, I hadn't considered that.

Will that same error message be used if there is an explicit super.init(), followed a conditional with one arm that uses an initDone()?

Reply:
You mean like this?
```chpl
if foo then
  super.init();
else
  this.initDone();
```

If so, then yes: that same error message will be used.

Reply:
I meant

```chapel
super.init();
if foo then
   ...;
else
  this.initDone();
```

in which case the error message would be "suggesting" this.init() when I think it's not allowed?

(Or a field initialization before the conditional.)

Reply:
Ah, I see. Good catch! I'll put it on the growing list of error message improvements 😄 

I plan on improving error messages over the next couple of weeks, so hopefully it will be resolved/improved soon. Feel free to create an issue if you like, but don't feel the need to submit a test case just yet. I can do that as I dig into error messages.

Reply:
Another approach might be to make the error message concerned with phase transitions: "If one branch transitions from phase 1 to phase 2, then all need to."  (for my case), or "from phase 0 to phase 2" if one branch had a legal this.init() and the other branch had nothing.

In the world where this.initDone() can be inferred (I think I read that somewhere), it would have to be more nuanced about one branch requires a transition that another branch isn't compatible with, or something.